wps-cloudformation-template.yaml

```diff
@@ -585,6 +585,81 @@ Resources:
                 Effect: "Allow"
                 Action: ["ses:SendRawEmail"]
                 Resource: ["*"]
+  LambdaClearBucketFunction:
+    Type: "AWS::Lambda::Function"
+    Condition: CreateEphemeralBucket
+    Properties:
+      Code:
+        ZipFile: |
+          import json
+          import boto3
+          import os
+          import cfnresponse
+
+          def handler(event, context):
+               print event
+               print context
+               try:
+
+                 if event['RequestType'] == 'Delete':
+                   s3 = boto3.resource('s3')
+                   bucket = s3.Bucket(os.getenv('BUCKET'))
+                   for obj in bucket.objects.all():
+                     obj.delete()
+                 cfnresponse.send(event, context, cfnresponse.SUCCESS, {'Data': 'OK'}, 'CustomResourcePhysicalID')
+               except Exception as e:
+                 print(e)
+                 return cfnresponse.send(event, context, cfnresponse.FAILED, {'Data': 'OK'}, 'CustomResourcePhysicalID')
+
+      Description: "Used to put objects into S3."
+      FunctionName: !Sub 'ClearBucket-${AWS::StackName}'
+      Handler: 'index.handler'
+      Role: !GetAtt LambdaClearBucketFunctionRole.Arn
+      Runtime: python2.7
+      Timeout: 60
+      Environment:
+        Variables:
+          BUCKET: !Ref EphemeralS3Bucket
+  ClearBucket:
+    Type: "Custom::ClearBucket"
+    Condition: CreateEphemeralBucket
+    Properties:
+      ServiceToken: !GetAtt LambdaClearBucketFunction.Arn
+  LambdaClearBucketFunctionRole:
+      Type: 'AWS::IAM::Role'
+      Condition: CreateEphemeralBucket
+      Properties:
+        AssumeRolePolicyDocument:
+          Version: 2012-10-17
+          Statement:
+            - Effect: Allow
+              Principal:
+                Service:
+                  - lambda.amazonaws.com
+              Action:
+                - 'sts:AssumeRole'
+        Path: /
+        ManagedPolicyArns:
+          - 'arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole'
+        Policies:
+          - PolicyName: PublishVersion
+            PolicyDocument:
+              Version: 2012-10-17
+              Statement:
+                - Effect: Allow
+                  Action:
+                    - 'lambda:PublishVersion'
+                  Resource: '*'
+          - PolicyName: S3Access
+            PolicyDocument:
+              Version: 2012-10-17
+              Statement:
+                - Effect: "Allow"
+                  Action: "s3:*"
```

Comment:
We can improve the security by giving access to only get and delete action. We can do the same for the other custom resource.